internal/graph/check_fast_path.go

```
@@ -494,13 +502,15 @@ func (c *LocalChecker) resolveFastPath(ctx context.Context, leftChans []chan *it
 					if storage.IterIsDoneOrCancelled(err) {
 						break
 					}
-					return nil, err
+					lastErr = err
+					continue
 				}
 				if processUsersetMessage(t.GetObject(), leftSet, rightSet) {
 					msg.Iter.Stop()
 					res.Allowed = true
 					span.SetAttributes(attribute.Bool("allowed", true))
-					return res, ctx.Err()
+					lastErr = nil
```

why we are not returning the error, and returning nil

---

it appears that this is a successful path, regardless of the existence of an error on the context. I assume that we don't want to indicate an error if the error is meaningless at this point.

---

@senojj correct. To match the existing contract of swallowing errors of `allowed=true` this would be thrown away.